Source/UI/Screens/SignUpScreen.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from "react";
+import { Alert, StyleSheet, Text, TextInput, TouchableOpacity, View,Image } from 'react-native';
+import auth from '@react-native-firebase/auth';
+// TODO:Implement the toggle password visibility.
+/* The 'SignUpScreen' component renders a sign-up form for new users. It collects user information i.e its personal Email ID and password. It handles form submission to register a new user.*/
+const SignUpScreen = () => {
+  const [email, setemail] = useState('');
+  const [password, setpassword] = useState('');
+ //The 'onRegister' function handles user registration using Firebase Authentication. 
+  const onRegister = () => {
+    auth().createUserWithEmailAndPassword(email, password)
+    .then(() => {
+    Alert.alert('User account created ');
+  })
+  .catch(error => {
+    if (error.code === 'auth/email-already-in-use') {
```

Comment:
good to use switch in case of errors instead of if as we can have more cases in future